client/resource_group/controller/controller.go

```diff
@@ -91,14 +95,20 @@ type ResourceGroupsController struct {
 }
 
 // NewResourceGroupController returns a new ResourceGroupsController which impls ResourceGroupKVInterceptor
-func NewResourceGroupController(clientUniqueID uint64, provider ResourceGroupProvider, requestUnitConfig *RequestUnitConfig) (*ResourceGroupsController, error) {
-	// TODO: initialize `requestUnitConfig`` from the remote manager server.
-	var config *Config
-	if requestUnitConfig != nil {
-		config = generateConfig(requestUnitConfig)
-	} else {
-		config = DefaultConfig()
-	}
+func NewResourceGroupController(
+	ctx context.Context,
+	clientUniqueID uint64,
+	provider ResourceGroupProvider,
+	requestUnitConfig *RequestUnitConfig,
+) (*ResourceGroupsController, error) {
+	var err error
+	if requestUnitConfig == nil {
+		if requestUnitConfig, err = loadRequestUnitConfig(ctx, provider); err != nil {
+			return nil, err
+		}
+	}
+	// TODO: do we need to reload the config from remote server periodically?
```

Comment:
IMO, reload is not required for now, and the cluster control surface can restart tidb

Reply:
I just removed this TODO.